### PR TITLE
Network manager implementation (M1)

### PR DESCRIPTION
**Pull request by schaable**

### CODE REVIEW

File: v-next/hardhat/src/internal/builtin-plugins/network-manager/type-extensions.ts [hunk 1]
```diff
@@ -0,0 +1,164 @@
+import type { NetworkConnection, NetworkManager } from "./types.js";
```
Comment (alcuadrado) [inline v-next/hardhat/src/internal/builtin-plugins/network-manager/type-extensions.ts:1]: Just a tip: this file can also be split into multiple files.

Maybe something like: 

```
type-extensions/
  index.ts -- imports the rest
  config.ts
  hooks.ts
```

Comment (schaable) [reply v-next/hardhat/src/internal/builtin-plugins/network-manager/type-extensions.ts:1]: I like the suggestion, I'll take it 👍

**Comment by schaable** [comment]
TODO:
 - Implement `resolveUserConfig`
 - Add tests and fix `HttpProvider` tests.
 - Add an example to the `example-project`

**Comment by alcuadrado** [comment]
I'll make some progress on this PR while Luis is OOO. I rebased it on top of `core-in-hardhat-continuation`, and started working on simplifying it a bit and some TODOs.

**Comment by alcuadrado** [comment]
Merging this as Luis is currently OOO. He'll continue working on this once he gets back.

**Pull request merged**